Like/unlike - Checkpoint 8

bpresser123 opened the pull request:
@robmadden I followed the [checkpoint](https://www.bloc.io/users/brian-presser/checkpoints/1693?roadmap_section_id=123) and added the suggested code. I made additions to the `likes_controller`, `model` and its associated views. 

I'm not sure about how to add the routes for `likes`. The checkpoint shows the `like` route inside of a block under `resources :bookmarks`, which is already in a block for `topics`, so I left some code in comments. At this point, I get an error when going to a bookmark and don't see links for like/unlike. 

I'm not sure where to put the `render partial:` code for the like view. The checkpoint says it goes in an erb file? I put it in `views/likes/index.html.erb` for now. Let me know what you think. Thanks!

bpresser123 commented:
@robmadden - Latest [commit](https://github.com/bpresser123/blocmarks/pull/6/commits/36a9e6130cd80a23b8229f9438ead1676a0420a0)

robmadden commented:
> I'm not sure about how to add the routes for likes. The checkpoint shows the like route inside of a block under resources :bookmarks, which is already in a block for topics, so I left some code in comments. At this point, I get an error when going to a bookmark and don't see links for like/unlike.

If that's what the checkpoint shows, you should do that. They're just nesting the routes under bookmarks. I'm not sure what you mean by "I don't see links for like/unlike". You don't automatically get links in your application by adding routes to a Rails application, you get routes.

> I'm not sure where to put the render partial: code for the like view. The checkpoint says it goes in an erb file? I put it in views/likes/index.html.erb for now. Let me know what you think. Thanks!

Put the render partial wherever you want to render the partial from. I.E. if you want it to show up in the `index` page then put it there. Since liked/unlikes are related to bookmarks, I would highly recommend adding them to either the Bookmarks show view or Bookmarks index view.

bpresser123 commented:
@robmadden I know that routes won't give me links, that's something I would implement in my views. The routes were giving me issues when trying go to a bookmark and etc. I added the routes as described in the checkpoint in my latest [commit](https://github.com/bpresser123/blocmarks/pull/6/commits/e1192c5dc472a4444873ce5e4a4ffc5bf8a58b13). I also added the render partial to the bookmarks show view. 

When I try to push these updates to Heroku, I get this error:
```bash
remote:        Running: rake assets:precompile
remote:        rake aborted!
remote:        /tmp/build_a8c075a61fdabb0a863a742046b3cff6/vendor/bundle/ruby/2.2.0/gems/activerecord-4.2.7/lib/active_record/associations/builder/association.rb:64:in '`initialize'```

bpresser123 commented:
@robmadden I also get a similar error when trying to test these features running the rails server locally:
```bash
Exiting
/Users/brianpresser/.rvm/gems/ruby-2.3.0/gems/activerecord-4.2.7/lib/active_record/associations/builder/association.rb:64:in `initialize': undefined method `arity' for :likes:Symbol (NoMethodError)```
